src/mut/section.cpp

```diff
@@ -51,12 +51,13 @@ bool Section::isRoot() const
     }
 }
 
-const std::vector<std::shared_ptr<Section>> Section::children() const
+const std::vector<std::shared_ptr<Section>>& Section::children() const
 {
     try {
         return _morphology->_children.at(id());
     } catch (const std::out_of_range&) {
-        return std::vector<std::shared_ptr<Section>>();
+        static std::vector<std::shared_ptr<Section>> empty;
+        return empty;
```

Comment:
Two things here:
1) try/catch is expensive in c++, doing something like:
```if(id() < _morphology->_children.size()){
        return _morphology->_children[id()];
    } else {
        return std::vector<int>();
    }
}
```
Is faster and simpler.

2) Making the vector once may probably is outweighed by the bookkeeping of having to track whether it's been created.  IIRC, vector creation is just zeroing two (?) `size(void*)`;

Reply:
The patch use a static variable when `id()` is not found in order to return a const reference to avoid a copy in the general case.
You are right though, the exception handling is avoidable.

Reply:
Didn't realize the return was a ref - in that case, good call.

I have no idea how people do code review in github, there is no where near enough context to read a diff, and no way to jump from conversation view to the comment in question; and even then, I have to click to get more context.

I suppose I need enlightenment.